homeassistant/components/melcloud/climate.py

```diff
@@ -116,6 +141,30 @@ def name(self):
         """Return the display name of this entity."""
         return self._name
 
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the optional state attributes with device specific additions."""
+        attr = {}
+
+        vane_horizontal = self._device.vane_horizontal
```

Comment:
Please create a attribute for this.

Reply:
Could you elaborate on this? Unfortunately I have no idea what I should do here.

Reply:
Please create attributes in the constructor for these values e.g. self._horizontal_positions and self._vertical_positions. If no positions are available set them to None.

Reply:
Oh the climate entity should have these properties, shouldn't it? Could you point me towards an integration already doing config validation based on entity properties?

Reply:
Yes, exactly.
Unfortunately I cannot think of one at the moment, but I think it is important to validate based on these properties to ensure that only available positions are passed on.

Reply:
I can't find suitable examples with `grep -R`. It does not look like entity state is used in service parameter validation. I could find non-constant and non-literal usages of `vol.In` only in config flow where it does make sense.

Is it even possible to use entity state in platform level services of an integration? The entity scope is not available when the service is registered and the available values depend based on the target entity.

Reply:
No, it is not possible and you are absolutely right about that. Your approach looks fine as it is now.